src/Client.php

```diff
@@ -918,11 +923,17 @@ public function send(Request $request = null)
                 throw new Client\Exception\RuntimeException('Adapter does not support streaming');
             }
 
+            $this->streamHandle = null;
```

Comment:
I don't like that this code is relying on side effects

Reply:
Do you have a better suggestion?